hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/AbstractSerializationService.java

```diff
@@ -147,7 +153,7 @@
                 out.writeInt(partitionHash, BIG_ENDIAN);
             }
 
-            out.writeInt(serializer.getTypeId(), BIG_ENDIAN);
+            out.writeInt(serializer.getTypeId(), serializerTypeIdByteOrder);
```

Comment:
Can you add a note on the toObject:

 final int typeId = data.getType();
          
That here the type is also read using big endian.

Reply:
Right, we always read the top-level object's `serializerTypeId` as `BIG_ENDIAN`.

1. This PR changes `InternalSerializationService.toBytes(Object obj, int leftPadding, boolean insertPartitionHash)` to always use the configured byte-order (instead of always using `BIG_ENDIAN`)
2. `InternalSerializationService.toBytes(Object obj, int leftPadding, boolean insertPartitionHash)` is only used when the normal response optimization kicks in, to serialize the response's value (ie the object returned from `Operation.getResponse`). This is not the top-level object, so it must use the configured byte order. The top-level object is the `NormalResponse` object and this one's `serializerTypeId` is always written in `BIG_ENDIAN` [in `writeResponsePrologueBytes`](https://github.com/vbekiaris/hazelcast/blob/16bd03c7794322b5989d1bd5641c80352c1117cb/hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/OutboundResponseHandler.java#L190).